projects/bp-gallery/src/components/common/OverviewContainer.tsx

```diff
@@ -25,6 +48,19 @@ const OverviewContainer = ({
           allowScrollButtonsMobile
           value={tabIndex}
           onChange={(_, value) => {
+            if (overviewPosition === OverviewContainerPosition.ARCHIVE_VIEW) {
+              if (archiveID) {
+                setSelectedTabs({
+                  ...selectedTabs,
+                  [overviewPosition]: {
+                    ...selectedTabs[overviewPosition],
+                    [archiveID]: value as number,
+                  },
+                });
+              }
+            } else {
+              setSelectedTabs({ ...selectedTabs, [overviewPosition]: value as number });
+            }
             setTabIndex(value as number);
```

Comment:
To clarify @rheaSPK's idea, the suggested change. I don't know whether it's more readable that way, but it has one level of indentation less. I'd say it's up to you, what you think is better :)
```suggestion
            setTabIndex(value as number);
            if (overviewPosition !== OverviewContainerPosition.ARCHIVE_VIEW) {
              setSelectedTabs({ ...selectedTabs, [overviewPosition]: value as number });
              return;
            }
            if (archiveID) {
              setSelectedTabs({
                ...selectedTabs,
                [overviewPosition]: {
                  ...selectedTabs[overviewPosition],
                  [archiveID]: value as number,
                },
              });
            }
```

Reply:
maybe also move this code (except the `setTabIndex`) into a `useEffect` that depends on the `tabIndex`, so that `selectedTabs` is always up to date to the `tabIndex`